engine/include/cubos/engine/ui/text/text.hpp

```diff
@@ -0,0 +1,45 @@
+/// @file
+/// @brief Component @ref cubos::engine::UIText.
+/// @ingroup ui-text-plugin
+
+#pragma once
+
+#include <string>
+
+#include <glm/vec4.hpp>
+
+#include <cubos/core/gl/render_device.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/api.hpp>
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/assets/assets.hpp>
+#include <cubos/engine/font/atlas/atlas.hpp>
+#include <cubos/engine/font/font.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Component used to draw text on the UI.
+    ///
+    /// @ingroup ui-text-plugin
+    struct CUBOS_ENGINE_API UIText
+    {
+        CUBOS_REFLECT;
+
+        /// @brief The text of the element.
+        std::string text;
+
+        /// @brief The color of the text.
+        glm::vec4 color{1.0F};
+
+        /// @brief The font size to draw the characters.
+        float fontSize{24.0F};
+
+        /// @brief The font atlas to be used.
+        Asset<FontAtlas> fontAtlas{AnyAsset("ab32bedd-8ea5-49f0-bca6-0b319feb1207")};
+
+        core::gl::VertexArray vertexArray{nullptr};
+
+        size_t vertexCount{0};
```

Comment:
One interesting approach to hide these internal implementation data which is being used in the new audio plugin, is to have a separate, private (header in the `src/` directory) component called `UITextImpl` which contains these private fields.
Then you add observers which automatically add and remove that component when the main component is removed, and you no longer have the problem of the user being able to mess with them!
I think it is also a good fit here, what do you think?

Reply:
yes ok seems good to me will implement that change